powerdns.go

```diff
@@ -154,29 +154,41 @@ func (p *Client) do(req *http.Request, v interface{}) (*http.Response, error) {
 
 	if resp.StatusCode == 401 {
 		return resp, &Error{
-			Status:  resp.Status,
-			Message: "Unauthorized",
+			Status:     resp.Status,
+			StatusCode: resp.StatusCode,
+			Message:    "Unauthorized",
 		}
 	}
 
 	if resp.StatusCode < 200 || resp.StatusCode > 299 {
 		defer func() {
 			_ = resp.Body.Close()
 		}()
-
-		apiError := new(Error)
-		err = json.NewDecoder(resp.Body).Decode(apiError)
-		if err != nil {
-			return resp, err
+		var message string
+
+		if resp.Header.Get("Content-Type") == "application/json" {
+			apiError := new(Error)
+			err = json.NewDecoder(resp.Body).Decode(&apiError)
+			if err != nil {
+				return resp, err
+			}
+			message = apiError.Message
+		} else {
+			messageBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return resp, err
+			}
+			message = string(messageBytes)
```

Comment:
Please add unit tests for these changes.

CI reference: https://github.com/joeig/go-powerdns/pull/17/checks?check_run_id=622562184#step:5:208

Afterwards, you can check the test coverage easily with the following commands:
```
make test
go tool cover -html=c.out
```

Reply:
I have added two negative cases to cover wrong decoding. 
Also Test401Handling has been changed. It was not working correct because there is no mock for /servers.

Reply:
This looks pretty good now and the negative test cases work well in case the test cases are executed against the mocks.

However, I agree to the point that there is improvement potential with the mocks. Therefore I would encourage you to run the unit tests once more, but this time against the real endpoints.

You can start the Docker environment with `docker-compose -f docker-compose-v4.2.yml up`, wait a moment until PowerDNS finished initialising, and then run `make test-without-mocks`.

If this result differs from the mocked result, it might be useful to tweak the mocks a bit.

Reply:
Sorry for long response, 

- Removed encoding checks after conversion, therefore I removed tests that were checking wrong conversion. 
- Fixed tests behavior for non-mocks tests.

Reply:
Ready to review again. @joeig